File: android/personstate/src/main/kotlin/com/farosos/personstate/Scheduler.kt
package com.farosos.personstate

/**
 * Reloj/programador inyectable para que `PersonStateMachine` pueda testear
 * el timer de gracia y el timeout sin depender de hilos reales ni esperar
 * minutos reales. En producción, la app lo respalda con un
 * `Handler`/coroutine; en tests, un `Scheduler` de prueba avanza el tiempo
 * manualmente.
 */
interface SchedulerToken

interface Scheduler {
    fun schedule(afterSeconds: Double, action: () -> Unit): SchedulerToken
    fun cancel(token: SchedulerToken)
}

File: ios/Sources/PersonStateMachine/Scheduler.swift
import Foundation

/// Reloj/programador inyectable para que `PersonStateMachine` pueda testear
/// el timer de gracia y el timeout sin depender de `Foundation.Timer` ni
/// esperar minutos reales. En producción, la app lo respalda con
/// `DispatchQueue`/`Timer`; en tests, un `Scheduler` de prueba avanza el
/// tiempo manualmente.
public protocol SchedulerToken {}

public protocol Scheduler {
    @discardableResult
    func schedule(after seconds: TimeInterval, _ action: @escaping () -> Void) -> SchedulerToken
    func cancel(_ token: SchedulerToken)
}
